operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,129 @@ const (
+	ResourceGroup string `json:"resourceGroup"`
+
+	// name is the name of the disk encryption set that will be set on the default storage class.
+	// The value should consist of only alphanumberic characters,
+	// underscores (_), hyphens, and be at most 80 characters in length.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength:=80
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9\_-]+$`
+	Name string `json:"name"`
+}
+
+// AzureCSIDriverConfigSpec defines properties that can be configured for the Azure CSI driver.
+type AzureCSIDriverConfigSpec struct {
+	// diskEncryptionSet sets the cluster default storage class to encrypt volumes with a
+	// customer-managed encryption set, rather than the default platform-managed keys.
+	// +optional
+	DiskEncryptionSet *AzureDiskEncryptionSet `json:"diskEncryptionSet,omitempty"`
+}
+
+// GCPKMSKeyReference gathers required fields for looking up a GCP KMS Key
+type GCPKMSKeyReference struct {
+	// name is the name of the customer-managed encryption key to be used for disk encryption.
+	// The value should correspond to an existing KMS key and should
+	// consist of only alphanumeric characters, hyphens (-) and underscores (_),
+	// and be at most 63 characters in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9\_-]+$`
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// keyRing is the name of the KMS Key Ring which the KMS Key belongs to.
+	// The value should correspond to an existing KMS key ring and should
+	// consist of only alphanumeric characters, hyphens (-) and underscores (_),
+	// and be at most 63 characters in length.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9\_-]+$`
+	// +kubebuilder:validation:MinLength:=1
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Required
+	KeyRing string `json:"keyRing"`
+
+	// projectID is the ID of the Project in which the KMS Key Ring exists.
+	// It must be 6 to 30 lowercase letters, digits, or hyphens.
+	// It must start with a letter. Trailing hyphens are prohibited.
+	// +kubebuilder:validation:Pattern:=`^[a-z][a-z0-9-]+[a-z0-9]$`
+	// +kubebuilder:validation:MinLength:=6
+	// +kubebuilder:validation:MaxLength:=30
+	// +kubebuilder:validation:Required
+	ProjectID string `json:"projectID"`
+
+	// location is the GCP location in which the Key Ring exists.
+	// The value must match an existing GCP location, or "global".
+	// Defaults to global, if not set.
+	// +kubebuilder:validation:Pattern:=`^[a-zA-Z0-9\_-]+$`
+	// +optional
+	Location string `json:"location,omitempty"`
+}
+
+// GCPCSIDriverConfigSpec defines properties that can be configured for the GCP CSI driver.
+type GCPCSIDriverConfigSpec struct {
+	// kmsKey sets the cluster default storage class to encrypt volumes with customer-supplied
+	// encryption keys, rather than the default keys managed by GCP.
+	// +optional
```

Comment:
What happens if this isn't set?

Reply:
The comment indicates what happens when the field isn’t set: the default cloud managed key is used.